src/Kiota.Builder/CodeDOM/CodeType.cs

```diff
@@ -1,22 +1,20 @@
 ﻿using System;
 
-namespace Kiota.Builder.CodeDOM
+namespace Kiota.Builder.CodeDOM;
+public class CodeType : CodeTypeBase, ICloneable
 {
-    public class CodeType : CodeTypeBase, ICloneable
+    public CodeElement? TypeDefinition
     {
-        public CodeElement TypeDefinition
-        {
-            get;
-            set;
-        }
-        public bool IsExternal {get;set;}
+        get;
+        set;
+    }
+    public bool IsExternal {get;set;}
 
-        public override object Clone()
-        {
-            return new CodeType{
-                TypeDefinition = TypeDefinition,
-                IsExternal = IsExternal
-            }.BaseClone<CodeType>(this);
-        }
+    public override object Clone()
+    {
+        return new CodeType{
+            TypeDefinition = TypeDefinition,
```

Comment:
Shouldn't `Clone` perform a deep copy?
I'm wondering what the following code does to the original object.

```csharp
var cloned = myCodeType.Clone();
cloned.TypeDefinition.Name = "new name";
// what would myCodeType.TypeDefinition.Name be?
```

Reply:
This is a bit of an exception as the definition lives within the code dom tree and we don't want to introduce a fork in the tree. But good catch, going to add a comment.
Strings are value copy, not reference copy in dotnet.

Reply:
I know the CLI refiner uses the `Clone()` then update pattern. Maybe the `TypeDefinition` shouldn't be cloneable in this case? The behavior would be confusing to contributors.

> Strings are value copy, not reference copy in dotnet.
I understand. The issue here is, `TypeDefinition` is not value copy.
```csharp
#myCodeType is ref1
#mycodeType.TypeDefinition is ref2
var cloned = myCodeType.Clone();
#cloned is ref3
#cloned.TypeDefinition is still ref2
cloned.TypeDefinition.Name = "new name";
```
Does this scenario make sense? Maybe CodeType shouldn't be cloneable

Reply:
That also impact the element of the tree, and all the types that rely on that definition. If that's the desired behaviour, you're probably ok, Thanks for checking.

Reply:
That's not the implied behavior of clone types though. Since the definition of `ICloneable` is ambiguous, I'll resolve this. However, the recommendation is to not implement this interface for public types.

> ## [Notes to Implementers](https://learn.microsoft.com/en-us/dotnet/api/system.icloneable?view=net-7.0#notes-to-implementers)
> ...
>Because callers of [Clone()](https://learn.microsoft.com/en-us/dotnet/api/system.icloneable.clone?view=net-7.0#system-icloneable-clone) cannot depend on the method performing a predictable cloning operation, we recommend that [ICloneable](https://learn.microsoft.com/en-us/dotnet/api/system.icloneable?view=net-7.0) not be implemented in public APIs.